From 297b519ba6f6dbd9d2882952569f7e469aec6915 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Mon, 28 Nov 2016 19:56:22 -0500
Subject: [PATCH 058/268] Make sure fetch_nonce() returns a string (not bytes)

...otherwise we send a b'' around the nonce in the response with Python3.

Fixes GitLab issue #170
---
 pylib/packet.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 4ebf43b..ab4575b 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -1172,7 +1172,7 @@ class ControlSession:
         self.doquery(opcode=CTL_OP_REQ_NONCE)
         if not self.response.startswith(polybytes("nonce=")):
             raise ControlException(SERR_BADNONCE)
-        return self.response.strip()
+        return polystr(self.response.strip())
 
     def mrulist(self, variables=None, rawhook=None):
         "Retrieve MRU list data"
-- 
2.7.4

